x/rollup/tx/l1_data.go

```diff
@@ -0,0 +1,81 @@
+package tx
+
+import (
+	"fmt"
+	"math/big"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/rlp"
+	rollupkeeper "github.com/polymerdao/monomer/x/rollup/keeper"
+)
+
+// L1DataAnteHandler will consume gas to compensate the sequencer for posting
+// the transaction to Ethereum. The gas cost is calculated based on the Ecotone
+// upgrade and the sequencer is expected to post the transaction using blobs.
+func L1DataAnteHandler(ctx sdk.Context, tx sdk.Tx, rollupKeeper *rollupkeeper.Keeper) (sdk.Context, error) { //nolint:gocritic // hugeparam
```

Comment:
Should the behavior change if the tx is being simulated? We currently don't accept a `simulate` arg like other ante handlers, not sure if that's an issue.

Reply:
There shouldn't be any behavior difference for us during `simulate` mode since we're basing the gas consumption on the current L1 attributes and the size of the tx. The current L1 attributes could change between the simulated tx and the real tx and result in an out-of-gas error. However, the user is expected to account for that when setting their gas limit for the real tx (normally 1.1x or 1.2x of the gas used for the simulated tx).

 Also, `simulate` is passed into `DepositAnteHandler` but it's never used there.